lib/cog/command/pipeline/executor.ex

```diff
@@ -198,169 +181,107 @@ defmodule Cog.Command.Pipeline.Executor do
     end
   end
 
-  @doc """
-  `bind` -> {:next_state, :maybe_expand}
+  def plan_next_invocation(:timeout, %__MODULE__{invocations: [current_invocation|remaining],
+                                                 output: previous_output,
+                                                 user_permissions: permissions}=state) do
 
-  Binds the current invocation to the scope. The scope is created based on the
-  context. nil context: blank scope, map context: singular scope, list context:
-  multiple scopes
-  """
-  def bind(:timeout, %__MODULE__{current: current, context: context}=state) do
-    scope = get_scope(context)
-    |> resolve_scope(current)
-    case bind_scope(scope, current) do
+    # If a previous command generated output, we don't need to retain
+    # any templating information, because the current command now
+    # controls how the output will be presented
+    context = strip_templates(previous_output)
+
+    case Cog.Command.Pipeline.Planner.plan(current_invocation, context, permissions) do
+      {:ok, plans} ->
+        {:next_state, :execute_plan, %{state |
+                                       output: [],
+                                       current_plan: nil,
+                                       plans: plans,
+                                       invocations: remaining}, 0}
       {:error, {:not_found, var}} ->
         Helpers.send_reply("I can't find the variable '$#{var}'.", state.request, state.mq_conn)
```

Comment:
Outside the scope of this PR but we should settle on Cog's voice. Some errors are written in a formal somewhat detached tone whilst others use informal and familiar phrasing.